.travis.yml

```diff
@@ -1,25 +1,45 @@
+dist: trusty
+sudo: required
+
 language: php
 
+cache:
+    directories:
+        - $HOME/.composer/cache/files
+
+
+env:
+  global:
+    # For functional tests
+    - COMPOSE_FILE="doc/docker-compose/base-dev.yml:doc/docker-compose/selenium.yml"
+
 matrix:
+    fast_finish: true
     include:
         - php: 5.6
         - php: 7.0
         - php: 7.1
           env: CHECK_CS=true
+        # Functional
+        - php: 7.1
+          env: BEHAT_OPTS="--profile=rest --tags=~@broken --suite=fullJson" EZPLATFORM_BRANCH=1.13 SYMFONY_ENV=behat SYMFONY_HTTP_CACHE=1
+        - php: 7.1
+          env: BEHAT_OPTS="--profile=rest --tags=~@broken --suite=fullJson" EZPLATFORM_BRANCH=master SYMFONY_ENV=behat
 
 # test only master (+ Pull requests)
 branches:
     only:
         - master
 
 before_script:
-    - travis_retry composer selfupdate
     - echo "memory_limit=-1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
-    - travis_retry composer install --prefer-dist --no-interaction
+    - if [ "$BEHAT_OPTS" = "" ] ; then travis_retry composer selfupdate && travis_retry composer install --prefer-dist --no-interaction ; fi
+    - if [ "$BEHAT_OPTS" != "" ] ; then ./tests/.travis/prepare_behat.sh ; fi
```

Comment:
Could we change the name to not be behat ? We are used to it, but it might confuse people out :)